docs/_docs/usage.md

```diff
@@ -23,8 +23,8 @@ Here are some of the most common commands:
 
 * `jekyll new PATH` - Creates a new Jekyll site with default gem-based theme at specified path. The directories will be created as necessary.
 * `jekyll new PATH --blank` - Creates a new blank Jekyll site scaffold at specified path.
-* `jekyll build` or `jekyll b` - Performs a one off build your site to `./_site` (by default).
-* `jekyll serve` or `jekyll s` - Builds your site any time a source file changes and serves it locally.
+* `jekyll build` or `jekyll b` - Performs a one off build your site to `./_site`, using the production configuration (by default).
```

Comment:
There is no *production* configuration technically. `jekyll serve` explicitly tests if the `env == "development"` which is the default environment. `jekyll build` does no such checks.

```suggestion
* `jekyll build` or `jekyll b` - Performs a one off build your site to `./_site` (by default).
```

Reply:
Perhaps this phrasing, to remove the "production" word while still explaining _which_ configuration is used to do the build?

```suggestion
* `jekyll build` or `jekyll b` - Performs a one off build your site to `./_site`, using the configuration from `_config.yml` (by default).
```

Reply:
Perhaps you're misunderstanding or perhaps overthinking.
Stating `... configuration from '_config.yml ...' is redundant because there is no other configuration.

`jekyll build` executes a build in the context of the current directory with default settings. To override default settings, one may either use CLI flags like `--source`, `--config`, `--incremental`, etc or use their counterparts (if available) directly in the config file.
`jekyll serve` is just a superset of `jekyll build`. **It does everything `jekyll build` does verbatim but additionally overrides `site.config["url"]` and starts up a local webserver for the files in the destination directly.**

To summarize figuratively, the two commands can be see as:

```ruby
j_build = run_build_process
j_serve = j_build + override_site_url + start_server
```

Reply:
> jekyll serve is just a superset of jekyll build. It does everything jekyll build does verbatim but additionally overrides site.config["url"] and starts up a local webserver for the files in the destination directly.

`jekyll serve` _does not currently override_ `site.config["url"]`. This has been the case since https://github.com/jekyll/jekyll/pull/7253 . My PR reverts that PR, and fixes documentation, for the reasons described in this PR's description.

I'm being explicit about _where_ the build settings for `jekyll build` come from, and what the intended use case of `jekyll build` is in contrast to `jekyll serve`. The reason I am being so painstakingly explicit is because this whole exercise comes from someone believing that `jekyll serve` built deployable, production-site-ready code into `/_site` like `jekyll build` does.

The revised line is meant to be read in contrast with the next line in the diff:

```
* `jekyll serve` or `jekyll s` - Builds your site any time a source file changes and serves it locally, using placeholders for the site URLs to enable it to be served locally (by default). Should not be used to generate a site copy that will be deployed to a live website.
```

I'm trying to prevent future occurrences of a misconception which has already occurred and which has already led to negative impacts on Jekyll users. For further reading, please see https://github.com/jekyll/jekyll/pull/7253#issuecomment-777653085

Reply:
> `jekyll serve` _does not currently override_ `site.config["url"]`

Yes. I agree. But this PR is undoing that behavior. So the documentation should be based on the future mindset.
Anyways, all this back-n-forth is not breaking the stalemate.
I'd like to ship this change via v4.2.1 in the near future.

So here's my recommendation:
- Please undo the changes you have currently made to `docs/_docs/usage.md`. Let's keep those summaries concise.
- Find another suitable place (`_docs/usage.md` included) to document the bullet-points raised in [#7253 (comment)](#7253#issuecomment-777786105)
    > - describe the effects of `serve` upon the variables it treats differently
    > - state that `serve` serves from the same build directory used by `jekyll build`, and encourage users to run `jekyll build` before deploying files from that directory to a production server.

Reply:
These changes have been removed in https://github.com/jekyll/jekyll/pull/8620/commits/bf406f55ea6c48a655311a55a4475149f15c90cf

The bullet-points raised have been addressed in changes to the tutorial section. I thought about tacking them on at the bottom of the usage section, but that breaks the flow of the page, and the information would be better contained in the bullet points. Thus, addenda to the tutorial.